### PR TITLE
[Chat][Feature] Chat middleware and action handler tests

### PR DESCRIPTION
**Pull request by Mrayyan**
## Purpose

Unit tests for chat middleware and action handler

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open...
* Click on...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by ShaunaSong — DISMISSED** [review]

File: azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/helper/MainCoroutineRule.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.azure.android.communication.ui.chat.helper
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/helper/MainCoroutineRule.kt:1]: missing copyright

File: azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/redux/middleware/handler/ChatMiddlewareActionHandlerUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
+package com.azure.android.communication.ui.chat.redux.middleware.handler
+
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/redux/middleware/handler/ChatMiddlewareActionHandlerUnitTest.kt:2]: missing copyright

File: azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/redux/middleware/ChatMiddlewareUnitTest.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
+package com.azure.android.communication.ui.chat.redux.middleware
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/redux/middleware/ChatMiddlewareUnitTest.kt:1]: copyright

File: azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/ACSBaseTestCoroutine.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.azure.android.communication.ui.chat
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/ACSBaseTestCoroutine.kt:1]: copyright

File: azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/helper/MainCoroutineRule.kt (hunk 1 above)
Comment (Mrayyan) [reply azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/helper/MainCoroutineRule.kt:4]: Added copyright tags in new commit.

**Review by iaulakh — APPROVED** [review]

**Review by dhiraj — APPROVED** [review]

**Pull request merged**